core/templates/dev/head/pages/library/library.html

```diff
@@ -1,11 +1,7 @@
 {% extends 'pages/base.html' %}
 
 {% block maintitle %}
-  {% if page_mode in ['group', 'search'] %}
-    Find explorations to learn from - Oppia
-  {% else %}
-    I18N_LIBRARY_PAGE_TITLE
-  {% endif %}
+ I18N_LIBRARY_PAGE_TITLE
 {% endblock maintitle %}
```

Comment:
I think your main goal here is to replace `{% if page_mode in ['group', 'search'] %}`.
Yes, you removed it. However, the library title directive does not replace the removed Jinja logic.

To reproduce the problem:
1) Go to oppia.org/library, the initial title is `Exploration Library - Oppia`. 
2) Enter some keyword into the search bar and hit enter, the title is now `Find exploration in Oppia`

When I checked out your PR branch, the title remains the same when I switch to "search" mode.